client/OPUNetGameSelectWnd.cpp

```diff
@@ -894,3 +895,45 @@ void OPUNetGameSelectWnd::OnClickCancel()
 
 	EndDialog(this->hWnd, true);			// 0 = Start, 1 = Cancel
 }
+
+
+void OPUNetGameSelectWnd::CreateServerAddressToolTip()
+{
+	HMODULE globalHInstance = GetModuleHandle(nullptr);
+
+	HWND hwndToolTip = CreateWindowEx(WS_EX_TOPMOST, TOOLTIPS_CLASS, nullptr,
+		WS_POPUP | TTS_NOPREFIX | TTS_ALWAYSTIP,
+		CW_USEDEFAULT, CW_USEDEFAULT, CW_USEDEFAULT, CW_USEDEFAULT,
+		this->hWnd, nullptr, globalHInstance, nullptr);
+
+	std::string toolTipText = "Server Address is represented by either an IP Address or DNS Name.\r\n";
+	toolTipText += "May point to an instance of a NetFix Server or Outpost2.exe.\r\n";
+	toolTipText += "Appending a port number is optional in either format.\r\n";
+	toolTipText += "IP Address example: 192.168.1.2:47800\r\n";
+	toolTipText += "DNS example: outpost2.net:4780";
+
+	TOOLINFO toolInfo = { 0 };
+	toolInfo.cbSize = sizeof(TOOLINFO);
+	toolInfo.uFlags = TTF_SUBCLASS;
+	toolInfo.hwnd = this->hWnd;
+	toolInfo.hinst = globalHInstance;
+	toolInfo.lpszText = TEXT(const_cast<char*>(toolTipText.c_str()));
```

Comment:
This initialization could potentially be combined, though I'm fine with it staying unpacked like this if you feel it helps readability. By leaving it unpacked, the field names remain visible.

Reply:
The core of this function is from an MSDN article, which probably displayed it unpacked to assist readability as sample code. I haven't thought about it too much one way or the other, but will leave it unpacked for now.